### PR TITLE
Fix duplicate RBAC breaking External Metrics Provider

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

We were creating (attempting to create) a Rolebinding to consume from the CM `extension-apiserver-authentication-reader`.
 
### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: controllers/datadogagent/clusteragent.go [hunk 1]
```diff
@@ -1227,6 +1227,12 @@ func buildClusterAgentClusterRole(dda *datadoghqv1alpha1.DatadogAgent, name, age
 			Verbs:         []string{rbac.GetVerb, rbac.CreateVerb, rbac.UpdateVerb},
 		})
 
+		rbacRules = append(rbacRules, rbacv1.PolicyRule{
+			APIGroups: []string{rbac.AutoscalingK8sIoAPIGroup},
+			Resources: []string{rbac.VPAResource},
+			Verbs:     []string{rbac.ListVerb, rbac.GetVerb, rbac.WatchVerb},
```
[the remark this answers is not in the document]
Comment (CharlyF) [reply controllers/datadogagent/clusteragent.go:1233]: ah yes, that's for v8.x but you are right

**Review by clamoriniere — COMMENTED** [review]
it seems that VPA is missing from this liste: https://github.com/DataDog/datadog-operator/blob/main/controllers/datadogagent_controller.go#L122-L140

**Review by clamoriniere — COMMENTED** [review]
looks good. could you run the `make generate` command (or equivalent) to generate the RBAC in the kustomize folder, and OLM

**Comment by CharlyF** [comment]
> looks good. could you run the `make generate` command (or equivalent) to generate the RBAC in the kustomize folder, and OLM
@clamoriniere, these have not been updated in quite some time, I'd rather do that in a separate PR to unblock the external metrics provider.
sounds good?

**Review by levan-m — APPROVED** [review]

**Pull request merged**